plugins/wallabag.koplugin/main.lua

```diff
@@ -364,14 +372,24 @@ function Wallabag:download(article)
 end
 
 -- method: (mandatory) GET, POST, DELETE, PATCH, etc...
--- apiurl: (mandatory) excluding the server path
+-- apiurl: (mandatory) API call excluding the server path, or full URL to a file
 -- headers: defaults to auth if given nil value, provide all headers necessary if in use
 -- body: empty string if not needed
 -- filepath: downloads the file if provided, returns JSON otherwise
-function Wallabag:callAPI( method, apiurl, headers, body, filepath )
+function Wallabag:callAPI( method, apiurl, headers, body, filepath)
     local request, sink = {}, {}
-    local parsed = url.parse(self.server_url)
-    request["url"] = self.server_url .. apiurl
+    -- Is it an API call, or a regular file direct download?
+    if apiurl:sub(1, 1) == "/" then
+        -- API call, is of the form "/random/api/call", append the server url
+        server = self.server_url
+        parsed = url.parse(self.server_url)
+        request["url"] = self.server_url .. apiurl
```

Comment:
```suggestion
        request.url = self.server_url .. apiurl
```

Reply:
Hmm, so I understand that it's easier to read that way, but since the rest of the file uses this same syntax should I replace every other occurrence of this pattern as well? Wouldn't it be weird if I only changed the lines you pointed out?